tests/manage/monitoring/conftest.py

```diff
@@ -277,3 +277,59 @@ def stop_mon():
         assert check_old_mons_deleted, msg
 
     return measured_op
+
+
+@pytest.fixture
+def workload_stop_ceph_osd(measurement_dir):
+    """
+    Downscales Ceph osd deployment, measures the time when it was
+    downscaled and alerts that were triggered during this event.
+
+    Returns:
+        dict: Contains information about `start` and `stop` time for stopping
+            Ceph osd pod
+    """
+    oc = ocp.OCP(
+        kind=constants.DEPLOYMENT,
+        namespace=config.ENV_DATA['cluster_namespace']
+    )
+    osd_deployments = oc.get(selector=constants.OSD_APP_LABEL)['items']
+    osds = [
+        deployment['metadata']['name']
+        for deployment in osd_deployments
+    ]
+
+    # get osd deployments to stop, leave even number of osd
+    osd_to_stop = osds[-1]
+    logger.info(f"osd disks to stop: {osd_to_stop}")
+    logger.info(f"osd disks left to run: {osds[:-1]}")
+
+    def stop_osd():
+        """
+        Downscale Ceph osd deployments for 11 minutes. First 1 minutes
+        the alert CephOSDDiskNotResponding should be in 'Pending'. After 1 minute
+        the alert turns into 'Firing' state.        This configuration of osd can be observed in ceph-mixins which
+        are used in the project:
+            https://github.com/ceph/ceph-mixins/blob/d22afe8c0da34490cb77e52a202eefcf4f62a869/config.libsonnet#L21
+        There should be also CephClusterWarningState alert that takes 10
+        minutest to be firing.
+
+        Returns:
+            str: Names of downscaled deployments
+        """
+        # run_time of operation
+        run_time = 60 * 11
```

Comment:
could be parameterized?

Reply:
I don't think that it should be parametrized as it is specific for the alert to be triggered. What json did you mean? If you mean ceph-mixins configuration then I wouldn't do that as it is external upstream project. Also if we don't load the value dynamically and if the test starts to fail in CI due to change in times for status change then we can easy notice that change and resolve it.